### DISCUSSION TITLE
Reading Gripper position with RWSStateMachineInterface

### DISCUSSION THREAD

**Discussion by bhomaidan1990** [ros-industrial/abb_librws#165] [Q&A]
Hi,

Is there a proper way to read the Grippers positions with `RWSStateMachineInterface` using StateMachins-Addin rather than calling service to [GetRAPIDSymbol](https://github.com/ros-industrial/abb_robot_driver_interfaces/blob/master/abb_robot_msgs/srv/GetRAPIDSymbol.srv) to read a variable which stores `g_GetPos()` in RAPID? 

What I want is to get the Smart gripper position and publish it to the `/joint_states` to be seen by MoveIt. thanks in advance.

**Comment by Mahdiford** [comment]
@jontje I see the g_GetPos() method has been added to the EGM/RWS. (I had not made any changes to the main RAPID files. No custom module added.) I can get the g_GetPos printed on the Flex Pendant, but it is confusing how to publish them as a separate ROS topic like joint_states. Would you please suggest a solution to this problem?
